Update NNCF WC documentation (#27101)
openvinotoolkit/openvino: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst

  `Larger Group Size`: Results in faster inference and a smaller model, but might
  compromise accuracy.

> * ``ratio`` controls the ratio between INT4 and layers compressed to ``backup_mode`` in the model.

Reviewer: Let's fix `INT4` as well as it can be NF4 and others in the future. You can rephase it to:
```suggestion
* ``ratio`` controls the ratio between the layers compressed to the precision defined by ``mode`` and the rest of the layers that will be kept in the ``backup_mode`` in the optimized model.
```

  4 bits. The method can sometimes result in reduced accuracy when used with
  Dynamic Quantization of activations. Requires dataset.

> * ``gptq`` - boolean parameter that enables the GPTQ method for more accurate weight

Reviewer 2: I would say exactly "INT4 weight compression"

* ``all_layers`` - boolean parameter that enables INT4 weight quantization of all
  Fully-Connected and Embedding layers, including the first and last layers in the model.

* ``lora_correction`` - boolean parameter that enables the LoRA Correction Algorithm
  to further improve the accuracy of INT4 compressed models on top of other
  algorithms - AWQ and Scale Estimation.

* ``backup_mode`` - defines a backup precision for mixed-precision weight compression.
  There are three modes: INT8_ASYM, INT8_SYM, and NONE, which retains
  the original floating-point precision of the model weights (``INT8_ASYM`` is default value).

> It is possible to generate a synthetic dataset using the `nncf.data.generate_text_data` method for

Reviewer: It looks like a subsection is required for this topic.
```suggestion
**Use synthetic data for LLM weight compression**

It is possible to generate a synthetic dataset using the `nncf.data.generate_text_data` method for
```

data-aware weight compression. The method takes a language model (e.g. from `optimum.intel.openvino`)
and a tokenizer (e.g. from `transformers`) as input and returns the list of strings generated by the model.
Note that dataset generation takes time and depends on various conditions, like the model size,
requested dataset length or environment setup. Also, since the dataset is generated by the model output,
it does not guarantee significant accuracy improvement after compression. This method is recommended
only when a better dataset is not available. Refer to the
`example <https://github.com/openvinotoolkit/nncf/tree/develop/examples/llm_compression/openvino/tiny_llama_synthetic_data>`__
for details of the usage.

  .. code-block:: python

>     from nncf import compress_weights, CompressWeightsMode, Dataset

Reviewer 3: ```suggestion
    from nncf import Dataset
```

Author: thanks!

* ``all_layers`` - boolean parameter that enables INT4 weight quantization of all
  Fully-Connected and Embedding layers, including the first and last layers in the model.

* ``lora_correction`` - boolean parameter that enables the LoRA Correction Algorithm
  to further improve the accuracy of INT4 compressed models on top of other
  algorithms - AWQ and Scale Estimation.

* ``backup_mode`` - defines a backup precision for mixed-precision weight compression.
  There are three modes: INT8_ASYM, INT8_SYM, and NONE, which retains
  the original floating-point precision of the model weights (``INT8_ASYM`` is default value).

> 
> **Use synthetic data for LLM weight compression**

Reviewer 4: ```suggestion

**Use synthetic data for LLM weight compression**
```

It is possible to generate a synthetic dataset using the `nncf.data.generate_text_data` method for
data-aware weight compression. The method takes a language model (e.g. from `optimum.intel.openvino`)
and a tokenizer (e.g. from `transformers`) as input and returns the list of strings generated by the model.
Note that dataset generation takes time and depends on various conditions, like the model size,
requested dataset length or environment setup. Also, since the dataset is generated by the model output,
it does not guarantee significant accuracy improvement after compression. This method is recommended
only when a better dataset is not available. Refer to the
`example <https://github.com/openvinotoolkit/nncf/tree/develop/examples/llm_compression/openvino/tiny_llama_synthetic_data>`__
for details of the usage.

>   .. code-block:: python
> 
>     from nncf import Dataset
>     from nncf.data import generate_text_data
> 
>     # Example: Generating synthetic dataset
>     synthetic_data = generate_text_data(model, tokenizer)
>     nncf_dataset = nncf.Dataset(synthetic_data, transform_fn)

Reviewer 4: ```suggestion
.. code-block:: python

   from nncf import Dataset
   from nncf.data import generate_text_data

   # Example: Generating synthetic dataset
   synthetic_data = generate_text_data(model, tokenizer)
   nncf_dataset = nncf.Dataset(synthetic_data, transform_fn)
```
